src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -93,21 +88,21 @@ private static bool IsValueWriteAtomic()
             //
             // See http://www.ecma-international.org/publications/files/ECMA-ST/Ecma-335.pdf
             //
+            Type valueType = typeof(TValue);
             bool isAtomic =
-                isClass
-                || valueType == typeof(Boolean)
-                || valueType == typeof(Char)
-                || valueType == typeof(Byte)
-                || valueType == typeof(SByte)
-                || valueType == typeof(Int16)
-                || valueType == typeof(UInt16)
-                || valueType == typeof(Int32)
-                || valueType == typeof(UInt32)
-                || valueType == typeof(Single);
+                   valueType == typeof(bool)
+                || valueType == typeof(char)
+                || valueType == typeof(byte)
+                || valueType == typeof(sbyte)
+                || valueType == typeof(short)
+                || valueType == typeof(ushort)
+                || valueType == typeof(int)
+                || valueType == typeof(uint)
+                || valueType == typeof(float);
 
             if (!isAtomic && IntPtr.Size == 8)
             {
-                isAtomic |= valueType == typeof(Double) || valueType == typeof(Int64);
+                isAtomic = valueType == typeof(double) || valueType == typeof(long);
```

Comment:
Is it a concern that the old code *added* double and long to the set of types tested above, while the new code *only* considers double and long as atomic, but not any of the smaller-sized types? If value size is the deciding factor, shouldn't `ulong` (and perhaps IntPtr and UIntPtr) be included in this test?